CONTRIBUTING.md

```diff
@@ -0,0 +1,146 @@
+    do_something_else();
+}
+
+do {
+    do_something();
+} while (cond);
+
+while (cond) {
+    do_something();   
+}
+
+switch (expression) {
+case 'constant-expression':
+    ...
+case 'constant-expression':
+    ...
+default:
+    ...
+}
+
+```
+
+* In general, do not use `goto`. The goto statement only comes in handy when a
+  function exits from multiple locations and some common work such as cleanup
+  has to be done. If there is no cleanup needed then just return directly.
+
+* For function, `static` and return type are put on the same line. Function's name is put on a
+  line of its own.
+
+```c
+static tlog_grc
+tlog_play_run(struct tlog_errs **perrs, int *psignal)
+{
+              ...
+}
+```
+
+* For regular comments, use C style (`/*` `*/`) comments, even for one-line.
+
+* For doc-comments (Doxygen comments), use the `/**` `*/` style featured.
+
+* For wrapping (`\`) of Macro, make sure to follow:
+
+```c
+#define CMP_BOOL(_name) \
+    do {                                                        \
+        if (_name != t._name##_out) {                           \
+            FAIL(#_name " %s != %s",                            \
+                 BOOL_STR(_name), BOOL_STR(t._name##_out));     \
+        }                                                       \
+    } while (0)
+```
+
+## Directories
+Tlog source is organized as follows.
+
+```
+.                                           <- Top-level build and documentation files
+├── doc                                     <- Documentation.
+├── include                                 <- C headers.
+│   ├── tlog
+│   ├── tltest
+│   │   ├── json_sink.h
+│   │   ├── json_source.h
+│   │   ├── json_stream_enc.h
+│   │   ├── misc.h
+├── lib                                     <- C library sources.
+│   ├── tlog
+│   ├── tltest
+|   │   ├── json_sink.c
+|   │   ├── json_source.c
+|   │   ├── json_stream_enc.c
+|   │   ├── misc.c
+├── src                                     <- Program sources.
+│   ├── tlog
+│   │   ├── tlog-play.c
+│   │   ├── tlog-rec.c
+│   │   ├── tlog-rec-session.c
+│   │   ├── tlog-prog.conf.m4
+│   ├── tltest
+│   │   ├── tltest-fd-json-reader.c
+│   │   ├── tltest-grc.c
+│   │   ├── tltest-json-esc.c
+│   │   ├── tltest-json-overlay.c
+│   │   ├── tltest-json-passthrough.c
+│   │   ├── tltest-json-sink.c
+│   │   ├── tltest-json-source.c
+│   │   ├── tltest-json-stream-btoa.c
+│   │   ├── tltest-json-stream-enc-bin.c
+│   │   ├── tltest-json-stream-enc-txt.c
+│   │   ├── tltest-json-stream.c
+│   │   ├── tltest-timespec.c
+│   │   ├── tltest-timestr.c
+├── m4                                      <- m4 macro sources.
+│   ├── autotools
+│   ├── tlog
+└── man                                     <- Manpages.
+```
+
+Follow instructions in [README](README.md) for how to compile and run test code.
+
+## License
+By contributing to Tlog, you agree that your contributions will be licensed
+under the [GNU GPLv2](COPYING) file in the root directory of this source tree.
```

Comment:
The contributions won't be licensed under a *file*, but under a license. In particular the GNU General Public License, or "GNU GPL". Note how "GPL" already contains the word "License". Please fix.